hack/jenkins/e2e-runner.sh

```diff
@@ -21,96 +21,124 @@ set -o nounset
-        fi
-        # At this point, we want to have the following vars set:
-        # - bucket
-        # - build_version
-        gsutil -mq cp \
-            "gs://kubernetes-release/${bucket}/${build_version}/kubernetes.tar.gz" \
-            "gs://kubernetes-release/${bucket}/${build_version}/kubernetes-test.tar.gz" \
-            .
+if [[ "${JENKINS_USE_EXISTING_BINARIES:-}" =~ ^[yY]$ ]]; then
+    echo "Tolerating dirty workspace (because JENKINS_USE_EXISTING_BINARIES)"
+elif [[ "${JENKINS_TOLERATE_DIRTY_WORKSPACE:-}" =~ ^[yY]$ ]]; then
+    echo "Tolerating dirty workspace (because JENKINS_TOLERATE_DIRTY_WORKSPACE)"
+else
+    # .config and its children are created by the gcloud call that we use to
+    # get the GCE service account.
+    # console-log.txt is created by Jenkins, but is usually not flushed out
+    # this early in the script.
+    #
+    # TODO(ihmccreery) we haven't gotten any service accounts yet, so we should
+    # be able to remove the .config check
+    if [[ $(find . -not -path "./.config*" -not -name "console-log.txt" \
+              | wc -l) != 1 ]]; then
+        echo "${PWD} not empty, bailing!"
+        find .
+        exit 1
     fi
+fi
 
-    if [[ ! "${CIRCLECI:-}" == "true" ]]; then
-        # Copy GCE keys so we don't keep cycling them.
-        # To set this up, you must know the <project>, <zone>, and <instance>
-        # on which your jenkins jobs are running. Then do:
-        #
-        # # SSH from your computer into the instance.
-        # $ gcloud compute ssh --project="<prj>" ssh --zone="<zone>" <instance>
-        #
-        # # Generate a key by ssh'ing from the instance into itself, then exit.
-        # $ gcloud compute ssh --project="<prj>" ssh --zone="<zone>" <instance>
-        # $ ^D
-        #
-        # # Copy the keys to the desired location (e.g. /var/lib/jenkins/gce_keys/).
-        # $ sudo mkdir -p /var/lib/jenkins/gce_keys/
-        # $ sudo cp ~/.ssh/google_compute_engine /var/lib/jenkins/gce_keys/
-        # $ sudo cp ~/.ssh/google_compute_engine.pub /var/lib/jenkins/gce_keys/
-        #
-        # # Move the permissions for the keys to Jenkins.
-        # $ sudo chown -R jenkins /var/lib/jenkins/gce_keys/
-        # $ sudo chgrp -R jenkins /var/lib/jenkins/gce_keys/
-        if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
-            echo "Skipping SSH key copying for AWS"
-        else
-            mkdir -p ${WORKSPACE}/.ssh/
-            cp /var/lib/jenkins/gce_keys/google_compute_engine ${WORKSPACE}/.ssh/
-            cp /var/lib/jenkins/gce_keys/google_compute_engine.pub ${WORKSPACE}/.ssh/
-        fi
-    fi
+# We get the Kubernetes tarballs unless we are going to use old ones
+if [[ "${JENKINS_USE_EXISTING_BINARIES:-}" =~ ^[yY]$ ]]; then
+    echo "Using existing binaries; not cleaning, fetching, or unpacking new ones."
+elif [[ "${KUBE_RUN_FROM_OUTPUT:-}" =~ ^[yY]$ ]]; then
+    # TODO(spxtr) This should probably be JENKINS_USE_BINARIES_FROM_OUTPUT,
```

Comment:
K.  That's a change that's blocked on getting the PR builder under version control, so I'll leave this here.